--- rpm-build-macros-1.647/rpm.macros~	2012-03-13 18:09:42.876512532 +0200
+++ rpm-build-macros-1.647/rpm.macros	2012-03-13 18:10:01.389010815 +0200
@@ -2166,79 +2166,17 @@
 # BuildRequires:	rpmbuild(macros) >= 1.647
 # Requires:	systemd-units >= 38
 
-%systemd_reload /bin/systemd_booted && SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog /bin/systemctl --quiet daemon-reload || :
-
-%systemd_service_enable() \
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	/bin/systemctl --quiet enable %{*} || : \
-	/bin/systemd_booted && echo 'Run "/bin/systemctl start %{*}" to start %{*}.' || : \
-%{nil}
-
-%systemd_service_disable() \
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	/bin/systemctl --quiet disable %{*} || : \
-%{nil}
-
-%systemd_service() \
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	/bin/systemd_booted && /bin/systemctl --quiet %{*} || : \
-%{nil}
-
-%systemd_service_start() \
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	/bin/systemd_booted && /bin/systemctl --quiet start %{*} || : \
-%{nil}
-
-%systemd_service_stop() \
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	/bin/systemd_booted && /bin/systemctl --quiet stop %{*} || : \
-%{nil}
-
-%systemd_service_restart() \
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	/bin/systemd_booted && /bin/systemctl --quiet try-restart %{*} || : \
-%{nil}
-
-%systemd_service_reload() \
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	/bin/systemd_booted && /bin/systemctl --quiet reload %{*} || : \
-%{nil}
-
-%systemd_trigger() \
-want_enable_service() { \
-	[ -f /etc/sysconfig/rpm ] && . /etc/sysconfig/rpm \
-	[ ${RPM_ENABLE_SYSTEMD_SERVICE:-yes} = no ] && return 1 \
-	/sbin/chkconfig --list ${1%.service} 2>/dev/null | grep -qsv "[0-6]:on" && return 1 \
-	return 0 \
-}\
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	for s in %{*}; do \
-		if want_enable_service $s; then \
-			/bin/systemctl --quiet enable $s || : \
-		fi \
-	done \
-%{nil}
-
-%systemd_post() \
-	export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-	/bin/systemd_booted && /bin/systemctl --quiet daemon-reload || : \
-	if [ $1 -eq 1 ]; then \
-		/bin/systemctl --quiet enable %{*} || : \
-		/bin/systemd_booted && echo 'Run "/bin/systemctl start %{*}" to start %{*}.' || : \
-	elif [ -z "$NORESTART" ]; then \
-		/bin/systemd_booted && /bin/systemctl --quiet try-restart %{*} || : \
-	else \
-		/bin/systemd_booted && echo 'Run "/bin/systemctl restart %{*}" to restart %{*}.' || : \
-	fi \
-%{nil}
-
-%systemd_preun() \
-	if [ $1 -eq 0 ]; then \
-		export SYSTEMD_LOG_LEVEL=warning SYSTEMD_LOG_TARGET=syslog \
-		/bin/systemd_booted && /bin/systemctl --quiet stop %{*} || : \
-		/bin/systemctl --quiet disable %{*} || : \
-	fi \
-%{nil}
+%systemd_reload %{nil}
+%systemd_service_enable() %{nil}
+%systemd_service_disable() %{nil}
+%systemd_service() %{nil}
+%systemd_service_start() %{nil}
+%systemd_service_stop() %{nil}
+%systemd_service_restart() %{nil}
+%systemd_service_reload() %{nil}
+%systemd_trigger() %{nil}
+%systemd_post() %{nil}
+%systemd_preun() %{nil}
 
 %systemdunitdir /lib/systemd/system
 %systemdtmpfilesdir /usr/lib/tmpfiles.d
