--- ./rpm.macros~	2008-08-12 20:54:19.000000000 +0300
+++ ./rpm.macros	2008-08-18 22:58:42.471609996 +0300
@@ -764,9 +764,14 @@
 # remove python sources, so that check-files won't complain
 # (idea by glen <at> pld-linux <dot> org)
 %py_postclean() \
+%{expand:%%define __sip_%{?_enable_debug_packages} 1} \
+%{?__sip_1:> debugfiles-add.list;}\
 for d in %{py_sitescriptdir} %{py_sitedir} %*; do \
- [ ! -d "$RPM_BUILD_ROOT$d" ] || find "$RPM_BUILD_ROOT$d" -name '*.py' -print0 | xargs -0r -l512 rm;\
+	[ ! -d "$RPM_BUILD_ROOT$d" ] || find "$RPM_BUILD_ROOT$d" -name '*.py' \\\
+	%{?__sip_1:>> debugfiles-add.list;}\
+	%{!?__sip_1:-print0 | xargs -0r %{__rm} -f;}\
 done \
+%{expand:%%undefine __sip_%{?_enable_debug_packages}}\
 %{nil}
 
 # depmod macro
