--- rpm-build-macros-1.387/rpm.macros~	2007-07-04 00:59:27.123949791 +0300
+++ rpm-build-macros-1.387/rpm.macros	2007-07-04 01:06:44.903953858 +0300
@@ -281,8 +281,10 @@
 # Requires name = version-release
 %requires_releq()		%(echo '%*' | LC_ALL="C" xargs -r rpm -q --qf 'Requires: %%{name} = %%{epoch}:%%{version}-%%{release}\\n' | sed -e 's/ (none):/ /' | grep -v "is not")
 
-%releq_kernel()		%((LC_ALL="C" rpm -qf --qf '%%{name} = %%{epoch}:%%{version}-%%{release}\\n' %{_kernelsrcdir}/include/linux/version.h 2>/dev/null || echo ERROR) | sed -e 's/ (none):/ /' | sed -e 's:-headers::' | grep -vE "(is not|no such)")
-%requires_releq_kernel(s:)	Requires%{-s:(%{-s*})}: %releq_kernel
+%releq_kernel_up()		%((LC_ALL="C" rpm -qf --qf '%%{name}-up = %%{epoch}:%%{version}-%%{release}\\n' %{_kernelsrcdir}/include/linux/version.h 2>/dev/null || echo ERROR) | sed -e 's/ (none):/ /' | sed -e 's:-headers::' | grep -vE "(is not|no such)")
+%releq_kernel_smp()		%((LC_ALL="C" rpm -qf --qf '%%{name}-smp = %%{epoch}:%%{version}-%%{release}\\n' %{_kernelsrcdir}/include/linux/version.h 2>/dev/null || echo ERROR) | sed -e 's/ (none):/ /' | sed -e 's:-headers::' | grep -vE "(is not|no such)")
+%requires_releq_kernel_up(s:)	Requires%{-s:(%{-s*})}: %releq_kernel_up
+%requires_releq_kernel_smp(s:)	Requires%{-s:(%{-s*})}: %releq_kernel_smp
 
 %requires_eq()		%(echo '%*' | LC_ALL="C" xargs -r rpm -q --qf 'Requires: %%{name} = %%{epoch}:%%{version}\\n' | sed -e 's/ (none):/ /' -e 's/ 0:/ /' | grep -v "is not")
 %requires_eq_to() %(LC_ALL="C" rpm -q --qf 'Requires: %1 = %%{epoch}:%%{version}\\n' %2 | sed -e 's/ (none):/ /' | grep -v "is not")
@@ -1167,9 +1169,9 @@
 compile() {															\
 	L="<"; [[ '%{*}' != *$L$L* ]] || PATCH_SH="set -x -e;$(cat)"	\
 	set -e -x														\
-	local cfgs='%{?with_dist_kernel:dist}%{!?with_dist_kernel:nondist}'	\
+	local cfgs='%{?with_dist_kernel:%{?with_smp: smp}%{?with_up: up}}%{!?with_dist_kernel: nondist}' \
 																	\
-for cfg in $cfgs; do												\
+for cfg in ${cfgs:-dist}; do                                        \
 	[ -r "%{_kernelsrcdir}/config-$cfg" ] || exit 1					\
 																	\
 	rm -rf o														\
@@ -1279,8 +1281,8 @@
 %define KernelD $RPM_BUILD_ROOT/lib/modules/%{_kernel_ver}			\
 %define ModprobeD $RPM_BUILD_ROOT%{_sysconfdir}/modprobe.d/%{_kernel_ver} \
 																	\
-install -d %{KernelD}/%{-d*}									\
-%{?-s:install -d %{ModprobeD}}								\
+install -d %{KernelD}{,smp}/%{-d*}									\
+%{?-s:install -d %{ModprobeD}{,smp}}								\
 																	\
 for MODULE in {%{-m*},}; do											\
 	[ -n "${MODULE}" ] || continue									\
@@ -1291,11 +1293,19 @@
 		%{?-s:echo "alias ${MNAME} ${MNAME}-%{-s*}"					\\\
 			>> %{ModprobeD}/%{-n*}.conf}							\
 	%else															\
-		install ${MODULE}-dist.ko										\\\
+	%if %{with up}													\
+		install ${MODULE}-up.ko										\\\
 			%{KernelD}/%{-d*}/${MNAME}%{-s:-%{-s*}}.ko				\
 		%{?-s:echo "alias ${MNAME} ${MNAME}-%{-s*}"					\\\
 			>> %{ModprobeD}/%{-n*}.conf}							\
 	%endif															\
+	%if %{with smp}													\
+		install ${MODULE}-smp.ko									\\\
+			%{KernelD}smp/%{-d*}/${MNAME}%{-s:-%{-s*}}.ko			\
+		%{?-s:echo "alias ${MNAME} ${MNAME}-%{-s*}"					\\\
+			>> %{ModprobeD}smp/%{-n*}.conf}							\
+	%endif															\
+	%endif															\
 done																\
 %{nil}
 
@@ -1326,6 +1336,10 @@
 done																\
 %{nil}
 
+# alias Th kernel macros to SMP kernel
+%releq_kernel %releq_kernel_smp
+%requires_releq_kernel %requires_releq_kernel_smp
+
 # patchset macros
 # Author: Elan Ruusamäe <glen@pld-linux.org>
 #
