--- ./macros.in.~1~	2005-06-22 10:38:30.000000000 +0300
+++ ./macros.in	2005-06-22 10:38:30.000000000 +0300
@@ -132,7 +132,10 @@
 unset LINGUAS ||:\
 %{nil}
 
-%install %{?_enable_debug_packages:%{?buildsubdir:%{debug_package}}}\
+%install \
+%if 0%{?_enable_debug_packages}\
+%{?buildsubdir:%{debug_package}}\
+%endif\
 %%install\
 LANG=C\
 export LANG\
@@ -296,7 +299,7 @@
     %{?verbose:set -x;} \
     if [ -d "$RPM_BUILD_ROOT" ]; then \
 	echo "Strip executable binaries, archives and shared object files."; \
-	filelist=`find $RPM_BUILD_ROOT -type f ! -regex ".*ld-[0-9.]*so.*" %{?_noautostrip:! -regex "%{_noautostrip}"}`; \
+	filelist=`find $RPM_BUILD_ROOT -type f ! -regex ".*ld-[0-9.]*so.*" ! -regex ".*/usr/lib[0-9]*/debug/.*\.debug" %{?_noautostrip:! -regex "%{_noautostrip}"}`; \
 	elfexelist=`echo $filelist | xargs -r file | \
 		awk '/ELF.*executable/ {print $1}' | cut -d: -f1`; \
 	elfsharedlist=`echo $filelist | xargs -r file | \
