--- rpm-build-macros-1.672/rpm.macros.orig	2013-08-18 17:59:05.046732903 +0200
+++ rpm-build-macros-1.672/rpm.macros	2013-08-18 18:25:39.536665989 +0200
@@ -878,7 +878,7 @@
 # %define _noautoreq_perl Apache::.*
 # BuildRequires:	rpmbuild(macros) >= 1.672
 # %define _noautoreq_java ClassDataVersion
-%__noauto_regexp_helper(p:) %(echo "%*" | awk 'BEGIN{RS=" "}{printf("%{-p*}\\\\\\(%%s\\\\\\) ", $1)}')
+%__noauto_regexp_helper(p:) %(echo "%*" | awk 'BEGIN{RS=" "}{printf("%{-p*}\\\\(%%s\\\\) ", $1)}')
 
 #%_noautocompressdoc	%{nil}
 #
